catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java

```diff
@@ -949,25 +963,38 @@ public Response addDocument(
     if (metacard == null) {
       metacard = new MetacardImpl();
     }
+
+    Set<AttributeDescriptor> missingDescriptors = new HashSet<>();
```

Comment:
❓Extract `private Set<AttributeDescriptor> setMetacardAttributes(metacard, attributes)`?
We could also include L977-982 and name extract `private Metacard overrideAttributes(metacard, attributes);`.